pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -568,33 +580,38 @@ protected void internalCreatePartitionedTopic(AsyncResponse asyncResponse, int n
             return;
         }
         validateNamespaceOperationAsync(topicName.getNamespaceObject(), NamespaceOperation.CREATE_TOPIC)
-                .thenRun(() -> {
-                    Policies policies = null;
-                    try {
-                        policies = getNamespacePolicies(namespaceName);
-                    } catch (RestException e) {
-                        if (e.getResponse().getStatus() != Status.NOT_FOUND.getStatusCode()) {
-                            throw e;
+                .thenCompose((__) -> getNamespacePoliciesAsync(namespaceName).exceptionally(ex -> {
+                    Throwable unwrapped = FutureUtil.unwrapCompletionException(ex);
+                    if (unwrapped instanceof RestException) {
+                        RestException re = (RestException) unwrapped;
+                        if (re.getResponse().getStatus() == Status.NOT_FOUND.getStatusCode()) {
```

Comment:
this could use the newer Java syntax, `unwrapped instanceof RestException re && re.getResponse().getStatus() == Status.NOT_FOUND.getStatusCode()`